designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -122,14 +185,161 @@ class UsageStatisticsReportsSettingsDeterminer(
+          val attachments = listOfActivities
+            .map { activity =>
+              activity.attachments
+            }
+            .map(_.length)
+            .sorted
+          val attachmentMedian  = attachments.get(attachments.length / 2).getOrElse(0)
+          val attachmentAverage = attachments.sum / attachments.length
+          val attachmentExist   = attachments.count(_ > 0)
+
+          //        Comment stats
+          val comments = listOfActivities
+            .map { activity =>
+              activity.comments
+            }
+            .map(_.length)
+            .sorted
+          val commentsMedian  = comments.get(comments.length / 2).getOrElse(0)
+          val commentsAverage = comments.sum / comments.length
+
+          Map(
+            "m_a" -> attachmentMedian,
+            "v_a" -> attachmentAverage,
+            "e_a" -> attachmentExist,
+            "m_c" -> commentsMedian,
+            "v_c" -> commentsAverage
+          ).mapValuesNow(_.toString)
+      }
+
+    val componentStats =
+      fetchComponentList().map {
+        case List() => Map.empty[String, String]
+        case componentList =>
+          val withoutFragments = componentList.filterNot(comp => comp.componentType == ComponentType.Fragment)
+          val byName           = withoutFragments.groupBy(_.name)
+
+          Map(
+            "c_n" -> byName.size.toString
+          )
+      }
+
+    val uptimeStats =
+      fetchScenarioActions().map {
+        case List() => Map.empty[String, String]
+        case listForScenarios =>
+          val uptimes = listForScenarios.map { listOfActions =>
+            val durationsOfDeployment = calculateUpTimeStats(listOfActions).filter(_.isDefined).sequence.get.sorted
+            //         return average running time for this scenario
+            if (durationsOfDeployment.nonEmpty) {
+              durationsOfDeployment.sum / durationsOfDeployment.length
+            } else {
+              0
+            }
+
+          }
+          val uptimeAverage = uptimes.sum / uptimes.length
+          val uptimeMedian  = uptimes.get(uptimes.length / 2).get
+          Map(
+            "m_u" -> uptimeMedian,
+            "v_u" -> uptimeAverage
+          ).mapValuesNow(_.toString)
+      }
+
+    // Combine all stats
+    queryParamsFromScenarios
+      .zipWith(activityQueryParams)(_ ++ _)
+      .zipWith(componentStats)(_ ++ _)
+      .zipWith(uptimeStats)(_ ++ _)
+      .zipWith(generalStats)(_ ++ _)
+  }
+
+  private def calculateUpTimeStats(processActions: List[ProcessAction]): List[Option[Long]] = {
+    val iter                    = processActions.reverseIterator
+    var hasStarted              = false
+    var lastActionTime: Instant = Instant.now
+    val listOfUptime = iter.map {
+      case ProcessAction(_, _, _, _, _, performedAt, ScenarioActionName.Deploy, _, _, _, _, _) =>
+        if (hasStarted) {
+          val uptime = performedAt.getEpochSecond - lastActionTime.getEpochSecond
```

Comment:
If lastActionTime is assinged to `now` I am sure that the uptime is below 0 because now is bigger than performedAt from past

Reply:
But the scenario would need to be started before by another `ProcessAction` which would overwrite `lastActionTime` as we go from the oldest `ProcessAction` to newest. But to be sure I can add to get max between the calculation and 0

Reply:
Doesn't apply anymore as we will gather statistics only for currently running scenarios